bootable-jar/runtime/src/main/java/org/wildfly/core/jar/runtime/BootableJar.java

```diff
@@ -331,6 +341,24 @@ public static void run(Path jbossHome, List<String> args, ModuleLoader moduleLoa
         // At this point we can configure JMX
         configureJMX(moduleClassLoader, bootableJar.log);
 
+        // Automatic loading of Security providers.
+        // Needed for logic that requires access to providers prior elytron subsystem is configured.
+        final ServiceLoader<Provider> providerServiceLoader = ServiceLoader.load(Provider.class, moduleClassLoader);
+        Iterator<Provider> iterator = providerServiceLoader.iterator();
+        SecurityManager sm = System.getSecurityManager();
+        for (;;) try {
+            if (! (iterator.hasNext())) break;
```

Comment:
Any reason not to just use:
```java
for (Provider provider : providerServiceLoader)
````

Reply:
If we are looping I think we do need to do it the way it was written in JBoss Modules, I believe David selected that pattern based on experiences around lazy loading of the classes and problems encountered during loading.